class.c

```diff
@@ -0,0 +1,1035 @@
+    CvOUTSIDE_SEQ(PL_compcv) = PL_cop_seqmax;
+}
+
+void
+Perl_class_prepare_method_parse(pTHX_ CV *cv)
+{
+    PERL_ARGS_ASSERT_CLASS_PREPARE_METHOD_PARSE;
+
+    assert(cv == PL_compcv);
+    assert(HvSTASH_IS_CLASS(PL_curstash));
+
+    /* We expect this to be at the start of sub parsing, so there won't be
+     * anything in the pad yet
+     */
+    assert(PL_comppad_name_fill == 0);
+
+    PADOFFSET padix;
+
+    padix = pad_add_name_pvs("$self", 0, NULL, NULL);
+    assert(padix == PADIX_SELF);
+    PERL_UNUSED_VAR(padix);
+
+    intro_my();
+
+    CvNOWARN_AMBIGUOUS_on(cv);
+    CvIsMETHOD_on(cv);
+}
+
+OP *
+Perl_class_wrap_method_body(pTHX_ OP *o)
+{
+    PERL_ARGS_ASSERT_CLASS_WRAP_METHOD_BODY;
+
+    if(!o)
+        return o;
+
+    PADNAMELIST *pnl = PadlistNAMES(CvPADLIST(PL_compcv));
+
+    AV *fieldmap = newAV();
+    UV max_fieldix = 0;
+    SAVEFREESV((SV *)fieldmap);
+
+    /* padix 0 == @_; padix 1 == $self. Start at 2 */
+    for(PADOFFSET padix = 2; padix <= PadnamelistMAX(pnl); padix++) {
+        PADNAME *pn = PadnamelistARRAY(pnl)[padix];
+        if(!pn || !PadnameIsFIELD(pn))
+            continue;
+
+        U32 fieldix = PadnameFIELDINFO(pn)->fieldix;
+        if(fieldix > max_fieldix)
+            max_fieldix = fieldix;
+
+        av_push(fieldmap, newSVuv(padix));
+        av_push(fieldmap, newSVuv(fieldix));
+    }
+
+    UNOP_AUX_item *aux = NULL;
+
+    if(av_count(fieldmap)) {
+        Newx(aux, 2 + av_count(fieldmap), UNOP_AUX_item);
+
+        UNOP_AUX_item *ap = aux;
+
+        (ap++)->uv = av_count(fieldmap) / 2;
+        (ap++)->uv = max_fieldix;
+
+        for(Size_t i = 0; i < av_count(fieldmap); i++)
+            (ap++)->uv = SvUV(AvARRAY(fieldmap)[i]);
+    }
+
+    /* If this is an empty method body then o will be an OP_STUB and not a
+     * list. This will confuse op_sibling_splice() */
+    if(o->op_type != OP_LINESEQ)
+        o = newLISTOP(OP_LINESEQ, 0, o, NULL);
+
+    op_sibling_splice(o, NULL, 0, newUNOP_AUX(OP_METHSTART, 0, NULL, aux));
+
+    return o;
+}
+
+void
+Perl_class_add_field(pTHX_ HV *stash, PADNAME *pn)
+{
+    PERL_ARGS_ASSERT_CLASS_ADD_FIELD;
+
+    assert(HvSTASH_IS_CLASS(stash));
+    struct xpvhv_aux *aux = HvAUX(stash);
+
+    PADOFFSET fieldix = aux->xhv_class_next_fieldix;
+    aux->xhv_class_next_fieldix++;
+
+    Newxz(PadnameFIELDINFO(pn), 1, struct padname_fieldinfo);
+    PadnameFLAGS(pn) |= PADNAMEf_FIELD;
+
+    PadnameFIELDINFO(pn)->fieldix = fieldix;
+    PadnameFIELDINFO(pn)->fieldstash = (HV *)SvREFCNT_inc(stash);
+
+    if(!aux->xhv_class_fields)
+        aux->xhv_class_fields = newPADNAMELIST(0);
+
+    padnamelist_store(aux->xhv_class_fields, PadnamelistMAX(aux->xhv_class_fields)+1, pn);
+    PadnameREFCNT_inc(pn);
+}
+
+static void
+apply_field_attribute_param(pTHX_ PADNAME *pn, SV *value)
+{
+    if(!value)
+        /* Default to name minus the sigil */
+        value = newSVpvn(PadnamePV(pn) + 1, PadnameLEN(pn) - 1);
```

Comment:
Found the problem purely by testing, then tracked down the probable cause.

```
tony@venus:.../git/perl6$ cat ../feature-class2.pl
use v5.36;
use utf8;
use feature 'class';

class Foo {
    field $Ł :param;
};

Foo->new("Ł" => 1);
tony@venus:.../git/perl6$ ./perl -Ilib ../feature-class2.pl 
class is experimental at ../feature-class2.pl line 5.
field is experimental at ../feature-class2.pl line 6.
Required parameter 'Ł' is missing for "Foo" constructor at ../feature-class2.pl line 9.
```

Reply:
Ahyes, that needs some `newSVpvn_utf8`. I'll pop that in.